ion-schema/src/system.rs

```diff
@@ -1144,13 +1145,34 @@ impl SchemaSystem {
             .load_schema(id, &mut TypeStore::default(), None)
     }
 
+    /// Constructs a new schema using provided ISL content.
+    pub fn new_schema(&mut self, schema_content: &[u8], id: &str) -> IonSchemaResult<Arc<Schema>> {
+        let elements = Element::read_all(schema_content)?;
+        let isl = self
+            .resolver
+            .isl_schema_from_elements(elements.into_iter(), id)?;
+        self.resolver
+            .schema_from_isl_schema(isl.version(), isl, &mut TypeStore::default(), None)
+    }
+
     /// Requests each of the provided [`DocumentAuthority`]s, in order, to get ISL model for the
     /// requested schema id until one successfully resolves it.
     /// If an authority throws an exception, resolution silently proceeds to the next authority.
     pub fn load_isl_schema<A: AsRef<str>>(&mut self, id: A) -> IonSchemaResult<IslSchema> {
         self.resolver.load_isl_schema(id, None)
     }
 
+    /// Constructs a new ISL model using provided ISL content.
+    pub fn new_isl_schema(
+        &mut self,
+        schema_content: &[u8],
+        id: &str,
+    ) -> IonSchemaResult<IslSchema> {
+        let elements = Element::read_all(schema_content)?;
+        self.resolver
+            .isl_schema_from_elements(elements.into_iter(), id)
+    }
```

Comment:
As discussed offline: since this returns an unresolved model of the desired schema, it might be possible to have it live somewhere else to avoid the naming confusion with the `new_schema` method above. If it's feasible, that move doesn't have to happen in this PR.

Reply:
I will create a separate PR for this as currently this involves getting `isl_schema_from_elements` that lives in the resolver.